src/compiler.ts

```diff
@@ -5735,7 +5738,17 @@ export class Compiler extends DiagnosticEmitter {
           thisType,
           Constraints.CONV_IMPLICIT | Constraints.IS_THIS
         );
-        let elementExpr = this.compileExpression(assert(indexExpression), Type.i32, Constraints.CONV_IMPLICIT);
+        let setterIndexType = setterInstance.signature.parameterTypes[0];
+        let getterIndexType = getterInstance.signature.parameterTypes[0];
+        if (!setterIndexType.equals(getterIndexType)) {
+          this.error(
+            DiagnosticCode.Index_signature_type_should_be_the_same_for_element_accessors_in_type_0,
+            valueExpression.range, classInstance.internalName
+          );
```

Comment:
Can this perhaps use `errorRelated` to show both the signature of the setter and getter? :)